lib/external-grading/queue-rabbitmq.js

```diff
@@ -0,0 +1,18 @@
+class RabbitMqQueue {
+    constructor(name) {
+        this.name = name;
+    }
+
+    async init(conn) {
+        this.ch = await conn.createChannel();
+    }
+
+    submitJob(message, callback) {
+        (async () => {
+            await this.ch.assertQueue(this.name, { durable: true });
+            this.ch.sendToQueue(this.name, new Buffer(JSON.stringify(message)), { persistent: true });
+        })().then(() => callback(null)).catch(callback);
```

Comment:
Not that I really mind, but I guess I find the try/catch version above a bit easier to follow.